src/multibody/geometry.hxx

```diff
@@ -31,30 +31,48 @@
 
 namespace se3
 {
+  inline GeometryData::GeometryData(const GeometryModel & modelGeom)
+    : oMg(modelGeom.ngeoms)
+
+#ifdef WITH_HPP_FCL   
+    , activeCollisionPairs(modelGeom.collisionPairs.size(), true)
+    , distanceRequest (true, 0, 0, fcl::GST_INDEP)
+    , distanceResults(modelGeom.collisionPairs.size())
+    , collisionRequest (1, false, false, 1, false, true, fcl::GST_INDEP)
+    , collisionResults(modelGeom.collisionPairs.size())
+    , radius()
+    , collisionPairIndex(-1)
+    , innerObjects()
+    , outerObjects()
+  {
+    collisionObjects.reserve(modelGeom.geometryObjects.size());
+    BOOST_FOREACH( const GeometryObject & geom, modelGeom.geometryObjects)
+      { collisionObjects.push_back
+          (fcl::CollisionObject(geom.fcl)); }
+    fillInnerOuterObjectMaps(modelGeom);
+  }
+#else
+  {}
+#endif // WITH_HPP_FCL   
 
-  inline GeomIndex GeometryModel::addGeometryObject(const GeometryObject& object)
+  inline GeomIndex GeometryModel::addGeometryObject(GeometryObject object,
+                                                    const Model & model)
   {
-    Index idx = (Index) (ngeoms ++);
+    assert( (object.parentFrame != -1) || (object.parentJoint != -1) );
+
+    if( object.parentJoint == -1 )  // if no parent joint, autofill the attribute.
+      object.parentJoint = model.frames[object.parentFrame].parent; 
+
+    assert( (object.parentFrame == -1) 
+            || (model.frames[object.parentFrame].type == se3::BODY)  );
+    assert( (object.parentFrame == -1) 
+            || (model.frames[object.parentFrame].parentJoint == object.parentJoint) );
```

Comment:
-1: right. I dont like unsigned int, and here is a new reason for that. 
I thought that we said that parentFrame would be set to -1 when unset. Did I misunderstood your proposition? I was here simply following the same idea. 
For this particular place, we could simply remove this automatic behaviour (set parentJoint if the user set it to -1). Right now, the automatic behaviour is only used in the unittest, and is easy to modify.

Thx for the second bug. I indeed did not compile in debug mode before pushing.